src/Azure.Functions.Cli/Actions/AzureActions/PublishFunctionAppAction.cs

```diff
@@ -469,23 +470,35 @@ private async Task SyncTriggers(Site functionApp)
             }, retryCount: 5);
         }
 
-        private async Task<bool> HandleElasticPremiumLinuxPublish(Site functionApp, Func<Task<Stream>> zipStreamFactory)
+        private async Task HandleElasticPremiumLinuxPublish(Site functionApp, Func<Task<Stream>> zipStreamFactory)
         {
+            // If Azure Files settings are present, we deploy similar to Linux App Service
+            if (functionApp.AzureAppSettings.ContainsKey(Constants.WebsiteContentAzureFileConnectionString)
+                && functionApp.AzureAppSettings.ContainsKey(Constants.WebsiteContentShared))
+            {
+                await HandleLinuxAppServicePublish(functionApp, zipStreamFactory);
+                return;
+            }
+
+            ColoredConsole.WriteLine(WarningColor($"Your Linux Premium function app does not have Azure Files App Settings" +
+                $" ('{Constants.WebsiteContentAzureFileConnectionString}', '{Constants.WebsiteContentShared}') configured. Azure Files setup is required to enable" +
+                $"persistent storage in Linux Premium function apps." +
+                $" For more info, see https://docs.microsoft.com/en-us/azure/azure-functions/functions-recover-storage-account#required-application-settings"));
```

Comment:
I think we'll need a better aka.ms link here. @anirudhgarg or @amamounelsayed, do either of you know if we have some proper doc describing the need for Azure Files settings in Linux Premium that we could create aka.ms link for? I couldn't find one, but wanted to check first.